metaflow/metadata/metadata.py

```diff
@@ -497,41 +496,76 @@ def _metadata_to_json(self, run_id, step_name, task_id, metadata):
             for datum in metadata
         ]
 
-    def _tags(self):
+    def _get_system_info_as_dict(self):
+        """This function drives:
+        - sticky system tags initialization
+        - task-level metadata generation
+        """
+        sys_info = dict()
         env = self._environment.get_environment_info()
-        tags = [
-            resolve_identity(),
-            "runtime:" + env["runtime"],
-            "python_version:" + env["python_version_code"],
-            "date:" + datetime.utcnow().strftime("%Y-%m-%d"),
-        ]
+        sys_info["runtime"] = env["runtime"]
+        sys_info["python_version"] = env["python_version_code"]
+        identity_type, identity_value = resolve_identity_as_tuple()
+        sys_info[identity_type] = identity_value
         if env["metaflow_version"]:
-            tags.append("metaflow_version:" + env["metaflow_version"])
+            sys_info["metaflow_version"] = env["metaflow_version"]
         if "metaflow_r_version" in env:
-            tags.append("metaflow_r_version:" + env["metaflow_r_version"])
+            sys_info["metaflow_r_version"] = env["metaflow_r_version"]
         if "r_version_code" in env:
-            tags.append("r_version:" + env["r_version_code"])
-        return tags
+            sys_info["r_version"] = env["r_version_code"]
+        return sys_info
+
+    def _get_system_info_as_tags(self):
+        return [
+            "{}:{}".format(k, v) for k, v in self._get_system_info_as_dict().items()
+        ]
 
-    def _register_code_package_metadata(self, run_id, step_name, task_id, attempt):
+    def _get_system_info_as_metadatum_list(self):
```

Comment:
Minor nit - can we move this logic inside `_register_system_metadata`? It seems that's the only place this method is being used.

Reply:
OK